packages/atlas-service/src/main.ts

```diff
@@ -62,102 +68,261 @@ export async function throwIfNotOk(
   throw err;
 }
 
-const MAX_REQUEST_SIZE = 10000;
+const AI_MAX_REQUEST_SIZE = 10000;
 
-const MIN_SAMPLE_DOCUMENTS = 1;
+const AI_MIN_SAMPLE_DOCUMENTS = 1;
 
 type MongoDBOIDCPluginLogger = Required<MongoDBOIDCPluginOptions>['logger'];
 
+type AtlasServiceAuthState =
+  // Instance was just created
+  | 'initial'
+  // Trying to restore state if it was persisted before
+  | 'restoring'
+  // Successfully got token from oidc-plugin
+  | 'authenticated'
+  // Token expired (and being refreshed at the moment)
+  | 'expired'
+  // Encountered an error while requesting token (either on sign in or refresh)
+  | 'unauthenticated';
+
+type SecretStore = {
+  getItem(key: string): Promise<string | undefined>;
+  setItem(key: string, value: string): Promise<void>;
+};
+
+const SECRET_STORE_KEY = 'AtlasLoginOIDCPluginState';
+
+const defaultSecretStore: SecretStore = {
+  async getItem(key: string) {
+    try {
+      const { appName } = getStoragePaths() ?? {};
+      if (
+        process.env.COMPASS_E2E_DISABLE_KEYCHAIN_USAGE === 'true' ||
+        !appName
+      ) {
+        throw new Error('Unsupported environment');
+      }
+      return (await keytar.getPassword(appName, key)) ?? undefined;
+    } catch {
+      return Promise.resolve(undefined);
```

Comment:
nit, feel free to ignore:

```suggestion
      return undefined;
```